### DISCUSSION TITLE
Equivalent Von Mises Stress Calculation for Random Vibration Analysis

### DISCUSSION THREAD

**Discussion by dle-bss** [ansys/pydpf-post#410] [General]
Howdy. I'm having trouble matching up an equivalent Von Mises stress calculation for a random vibration case in Ansys Mechanical vs. PyAnsys-Post. As a sanity check, I've confirmed that I can get matching directional deformations for the random vibration case, and I can get matching equivalent Von Mises stress for a separate static structural case.

I'm specifically calculating the equivalent Von Mises stress using `stress_eqv_von_mises_nodal()` for the 1-sigma displacement result (loadstep 3, substep 1). Compared to Ansys Mechanical, the resulting stress is the correct order of magnitude (1e7 Pa in this case), but with some error (on the order of 0.1e7 Pa).

I'm wondering what the stress calculation method is for `stress_eqv_von_mises_nodal()` for a `ModalMechanicalSimulation()` object. Could this discrepancy I'm seeing having anything to do with the Segalman-Fulcher algorithm? According to this video, it's what's employed in Ansys Mechanical to report equivalent Von Mises stress: https://www.youtube.com/watch?v=6GMuxQBP5-E

```python
import ansys.dpf.core as dpf
from ansys.dpf import post

rst_path = "path/to/file.rst"

simulation = post.ModalMechanicalSimulation(rst_path)

# set id 32 in this context is loadstep 3, substep 1 i.e. 1-sigma displacement
stress = simulation.stress_eqv_von_mises_nodal(set_ids=32)
max_stress = stress.max().array[0]

print(f'Max Nodal Eqv VM Stress = {max_stress}')
```

**Comment by PProfizi** [comment]
Hi @dle-bss,
There are indeed several methods used to compute the equivalent von mises stress depending on the analysis context.
Here, you are using the `ModalMechanicalSimulation` class which is to be used for purely modal analyses.
Hence its `stress_eqv_von_mises_nodal()` method does not use the Segalman formulation.
As of right now, we do not have in PyDPF-Post a specific interface/class to post-process Random Vibration analyses, yet what I can propose if all you need is to get the Segalman formulation of equivalent von mises stress (without correlation between modes) is to use a tool available in PyDPF-Core to do just that (see [here](https://dpf.docs.pyansys.com/version/stable/api/ansys.dpf.core.operators.invariant.segalman_von_mises_eqv.html)).
If that would answer your need I can help writing a script.
